Mask Sensitive Data in Airflow CLI Using SecretsMasker (Fixes #59844)

Alearner12 opened the pull request:
This PR enhances the security of the Airflow CLI by ensuring sensitive keys and values are properly redacted when listing connections and variables. The CLI now leverages Airflow’s built-in [SecretsMasker.redact()](vscode-file://vscode-app/c:/Users/siddh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) mechanism to prevent accidental exposure of secrets, while still allowing users to view values intentionally when required.

##Changes

Updated CLI logic to apply SecretsMasker.redact() when displaying connection and variable values.
Ensured masking is applied when --hide-sensitive is used and by default behavior.

Extended core.sensitive_var_conn_names configuration to include commonly sensitive keys such as:
secret_key jwt_secret fernet_key sql_alchemy_conn other security-related fields
Added and verified unit tests to confirm correct masking behavior.

##How to Test

1.Create connections and variables containing sensitive keys and values.
2.Run the following CLI commands:
    airflow connections list
    airflow connections list --show-values
    airflow connections list --hide-sensitive

    airflow variables list
    airflow variables list --show-values
    airflow variables list --hide-sensitive


##Verify the behavior:

    Sensitive values are masked by default
    --hide-sensitive always redacts sensitive data
    --show-values displays the actual values as expected 

Screenshots

<img width="1151" height="829" alt="Screenshot 2025-12-27 180404" src="https://github.com/user-attachments/assets/b38134ce-eb03-4006-acd8-11427de3fca8" />
<img width="1184" height="364" alt="Screenshot 2025-12-27 180423" src="https://github.com/user-attachments/assets/b6d9ad8d-d4a8-4c00-b560-3d9960e7f9c9" />

potiuk commented on airflow-core/src/airflow/cli/commands/variable_command.py:
```diff
@@ -57,19 +50,14 @@ def _variable_mapper(var: Variable, show_values: bool = True, hide_sensitive: bo
-    # Determine what to show for the value
     if not show_values:
-        # Hide value completely
         result["value"] = "***"
```
we shoud just not have values at all here -> just keys.

potiuk commented on airflow-core/src/airflow/cli/commands/connection_command.py:
```diff
@@ -39,26 +39,46 @@
+    if show_values:
+        if hide_sensitive:
+            # Use Airflow's secrets masker to redact sensitive values
+            result["password"] = redact(conn.password, name="password")
```
Look at `redact` -> you should not specify name, when you have dict, redact will recursively pass name from keys of the dictionary and replace all sensitive values automatically. name-None is default. So it is way better if you redact the whole "result" dictionary - this will be way simpler:

1) prepare dictionary
2) if hide_sensitive -> redact the dictionary

potiuk commented on airflow-core/src/airflow/cli/commands/variable_command.py:
```diff
@@ -35,17 +36,49 @@
+    
+    # Only include the value if show_values is True
+    if show_values:
+        if hide_sensitive:
```
same here - prepare dictrionary and redact it if sensitive.

potiuk commented on airflow-core/src/airflow/cli/commands/variable_command.py:
```diff
@@ -151,4 +184,4 @@ def variables_export(args):
     with args.file as varfile:
         json.dump(var_dict, varfile, sort_keys=True, indent=4)
-        print_export_output("Variables", var_dict, varfile)
+        print_export_output("Variables", var_dict, varfile)
```
Your editor is modifying EOF - you should disable it (also prek will fix it for you automatically)

potiuk commented on airflow-core/tests/unit/cli/commands/test_connection_command.py:
```diff
@@ -56,7 +56,7 @@ def test_cli_connection_get(self, stdout_capture):
                 self.parser.parse_args(["connections", "get", "google_cloud_default", "--output", "json"])
             )
-        assert "google-cloud-platform:///default" in capture.getvalue()
+        assert '"get_uri": "***"' in capture.getvalue()
```
This is wrong. "connections get" should ALWAYS return unredacted value. Redaction is only needed for `list` command - this is what we agreed in the devlist discussion.

potiuk commented on airflow-core/src/airflow/cli/commands/connection_command.py:
```diff
@@ -71,10 +91,15 @@ def connections_get(args):
         data=[conn],
         output=args.output,
-        mapper=_connection_mapper,
+        mapper=lambda c: _connection_mapper(c, show_values=show_values, hide_sensitive=hide_sensitive),
```
`connections get` should not be redacted. Only list.

Commit:
fix: mask sensitive data in CLI using SecretsMasker.redact()

- Use redact() on whole dictionary in _connection_mapper and _variable_mapper
- connections get: always shows unredacted values (intentional retrieval)
- connections list: masks by default, --show-values shows masked values
- variables list: only shows keys by default, --show-values shows masked values

Commit:
fix: add type casts for redact() to satisfy MyPy

Alearner12 committed:
Add --show-values and --hide-sensitive flags to connections/variables list CLI

Alearner12 committed:
Merge branch 'apache:main' into main

Alearner12 committed:
Fix redact calls to explicitly pass field names for proper masking
